nextcord/polls.py

```diff
@@ -0,0 +1,439 @@
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import TYPE_CHECKING, List, Optional, Union, cast
+
+from .emoji import Emoji
+from .enums import PollLayoutType, try_enum
+from .errors import InvalidArgument
+from .iterators import AnswerVotersIterator
+from .partial_emoji import PartialEmoji
+from .utils import MISSING, utcnow
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from .abc import Snowflake
+    from .message import Message
+    from .state import ConnectionState
+    from .types.emoji import PartialEmoji as PartialEmojiPayload
+    from .types.polls import (
+        Poll as PollData,
+        PollAnswer as PollAnswerPayload,
+        PollAnswerCount as PollAnswerCountPayload,
+        PollCreateRequest as PollCreateRequestPayload,
+        PollMedia as PollMediaPayload,
+        PollResults as PollResultsPayload,
+    )
+
+    # for reviewers: this typehint type is precedent in the nextcord codebase i.e.
+    # https://github.com/nextcord/nextcord/blob/master/nextcord/message.py#L68
+    EmojiInputType = Union[Emoji, PartialEmoji, str]
+
+
+def resolve_emoji(emoji: Union[Emoji, PartialEmoji]) -> PartialEmojiPayload:
+    if isinstance(emoji, Emoji):
+        return cast(PartialEmojiPayload, emoji._to_partial().to_dict())
+
+    if isinstance(emoji, PartialEmoji):
+        return cast(PartialEmojiPayload, emoji.to_dict())
+
+    raise InvalidArgument(f"Cannot parse `{emoji}` into an emoji to send.")
+
+
+class PollMedia:
+    """
+    A common object that backs both the question and answer.
+
+    .. versionadded:: 3.0
+
+    Attributes
+    ----------
+    text: Optional[:class:`str`]
+        The text of the media.
+    emoji: Optional[Union[:class:`PartialEmoji`, :class:`Emoji`]]
+        The emoji of the media.
+
+    """
+
+    __slots__ = (
+        "text",
+        "emoji",
+    )
+
+    def __init__(self, text: str, emoji: Optional[EmojiInputType] = None) -> None:
+        self.text: str = text
+        self.emoji: Optional[PartialEmoji | Emoji] = emoji  # type: ignore[reportAttributeAccessIssue]
+        if isinstance(emoji, str):
+            self.emoji: Optional[PartialEmoji | Emoji] = PartialEmoji.from_str(emoji)
+
+    def __repr__(self) -> str:
+        return f"<PollMedia text={self.text} emoji={self.emoji}>"
+
+    def to_dict(self) -> PollMediaPayload:
+        payload: PollMediaPayload = {"text": self.text}
+        if self.emoji:
+            payload["emoji"] = resolve_emoji(self.emoji)
+
+        return payload
+
+    @classmethod
+    def from_dict(cls, data: PollMediaPayload) -> Self:
+        emoji = PartialEmoji.from_dict(data["emoji"]) if "emoji" in data else None
+
+        return cls(text=data["text"], emoji=emoji)
+
+
+class PollAnswer:
+    """
```

Comment:
Could these docstrings be like `"""So` just like `PollAnswerCount`, in line with the other multiline docstrings.